src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts

```diff
@@ -89,14 +91,47 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
     this.actionTriggersWithAction.clear()
 
     this.actionTriggers.forEach(actionTrigger => {
-      const actionForTrigger: Tv2Action | undefined = this.actions.find(action => action.id === actionTrigger.actionId)
-      if (!actionForTrigger) {
+      let action: Tv2Action | undefined = this.actions.find(action => action.id === actionTrigger.actionId)
+      if (!action) {
         return
       }
-      this.actionTriggersWithAction.set(actionTrigger, actionForTrigger)
+      if (action.type === PlaceholderActionType.CONTENT) {
+        action = this.findActionForContentPlaceholderAction(action as Tv2ContentPlaceholderAction, actionTrigger)
+      }
+      if (!action) {
+        return
+      }
+      this.actionTriggersWithAction.set(actionTrigger, action)
     })
   }
 
+  private findActionForContentPlaceholderAction(placeholderAction: Tv2ContentPlaceholderAction, actionTrigger: ActionTrigger<KeyboardTriggerData>): Tv2Action | undefined {
+    const filterCallback: (action: Tv2Action) => boolean = this.getActionFilterCallbackForPlaceholderAction(placeholderAction)
+
+    const actionsForSegmentToSearch: Tv2Action[] = this.actions.filter(filterCallback).sort((a, b) => a.rank - b.rank)
+    const indexToSelect: number = actionTrigger.data.actionArguments as number // We know this is a number because of Tv2ContentPlaceholderAction.
+    if (indexToSelect > actionsForSegmentToSearch.length) {
+      return
+    }
+    return actionsForSegmentToSearch[indexToSelect - 1] // indexToSelect is one-indexed, so we need to subtract one to get the correct index.
+  }
+
+  private getActionFilterCallbackForPlaceholderAction(placeholderAction: Tv2ContentPlaceholderAction): (action: Tv2Action) => boolean {
+    switch (placeholderAction.metadata.scope) {
+      case PlaceholderActionScope.ON_AIR_SEGMENT: {
+        const onAirSegment: Segment | undefined = this.rundown?.segments.find(segment => segment.isOnAir)
+        if (!onAirSegment) {
+          break
+        }
+        return action => Math.floor(action.rank) === onAirSegment.rank && placeholderAction.metadata.allowedContentTypes.includes(action.metadata.contentType)
+      }
+      default: {
+        this.logger.warn(`PlaceholderScope ${placeholderAction.metadata.scope} is not supported.`)
```

Comment:
Should `PlaceholderScope` be two words (with "scope" instead) or is it in reference to the `PlaceholderActionScope`?

Reply:
It is in reference to `PlaceholderActionScope`, but I think there is a sound argument in having it as two words for the log message.